src/main/java/jp/t2v/xanadu/ops/StreamOps.java

```diff
@@ -42,4 +42,12 @@
         return StreamSupport.stream(new TakeWhileSpliterator<>(self.spliterator(), predicate), false);
     }
 
+    public <A, B, C> Stream<C> zipWith(final Stream<A> self, final Stream<B> other, final BiFunction<? super A, ? super B, ? extends C> f) {
+        return StreamSupport.stream(new ZipSpliterator<>(self.spliterator(), other.spliterator(), f), false);
+    }
+
+    public <A, B, C> Stream<C> zipWith(final Stream<A> self, final Iterable<B> other, final BiFunction<? super A, ? super B, ? extends C> f) {
```

Comment:
overload introduces a pitfall. ex cannot use method reference, etc.
Would you rename it to `zipWithI`?